source/geometries/Next100SiPM.cc

```diff
@@ -71,6 +72,35 @@ void Next100SiPM::Construct()
 
   GeometryBase::SetLogicalVolume(sipm_logic_vol);
 
+  // COATING /////////////////////////////////////////////
+
+  G4LogicalVolume* coating_logic_vol;
+  if (with_coating_) {
+    G4String coating_name = sipm_name + "_WLS";
+
+    G4double coating_thickn_  = 1. * micrometer;
+
+    G4Box* coating_solid_vol =
+      new G4Box(coating_name, sipm_width/2., sipm_length/2., coating_thickn_/2.);
+
+    G4Material* coating_mt_ = MaterialsList::TPB();
+    coating_mt_->SetMaterialPropertiesTable(OpticalMaterialProperties::TPB());
+
+    coating_logic_vol = new G4LogicalVolume(coating_solid_vol, coating_mt_, coating_name);
+
+    G4double coating_zpos = sipm_thickn/2. + coating_thickn_/2.;
```

Comment:
I see your point, I didn't realise. The thing is the parameter `sipm_thickn` defines the size of the encasing, that includes the sensor area and the window. I've located the coating on top of this encasing. If we extend `sipm_thickn` to `sipm_thickn` + `coating_thickn_`, we're making the encasing larger and we would need to locate the coating below this volumen (on top of the window). I'm not sure if this is realistic or not. Maybe a way to extend the mother sipm volumen (in case it's coated) is by using `sipm_thickn` + `coating_thickn_` instead of `sipm_thickn` in the `dimensions_` definition in line 66.  @paolafer, could you give us a hand here?

Reply:
Sure, I'll call you.

Reply:
I see now your point, @gondiaz. I've increased the `sipm_thickn` in case of coating, and I've corrected the position of the other volumes accordingly.